pkg/autoscaler/autoscaler.go

```diff
@@ -434,4 +449,72 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
 			framework.WaitForMachineSetDelete(client, ms)
 		}
 	})
+
+	It("It scales from zero", func() {
+
+		// Only run in platforms which support autoscaling from/to zero.
+		clusterInfra, err := framework.GetInfrastructure(client)
+		Expect(err).NotTo(HaveOccurred())
+
+		platform := clusterInfra.Status.PlatformStatus.Type
+		switch platform {
+		case configv1.AWSPlatformType, configv1.GCPPlatformType, configv1.AzurePlatformType:
+			glog.Infof("Platform is %v", platform)
+		default:
+			Skip(fmt.Sprintf("Platform %v does not support autoscaling from/to zero, skipping.", platform))
+		}
+
+		// Anything we create we must cleanup
+		cleanupObjects := map[string]runtime.Object{}
+		defer func() {
+			for name, obj := range cleanupObjects {
+				if err := deleteObject(name, obj); err != nil {
+					glog.Infof("[cleanup] error deleting object %q (%T): %v", name, obj, err)
+				}
+			}
+		}()
+
+		By("Creating a new MachineSet with 0 replicas")
+		machineSetParams := framework.BuildMachineSetParams(client, 0)
+		targetedNodeLabel := fmt.Sprintf("%v-scale-from-zero", autoscalerWorkerNodeRoleLabel)
+		machineSetParams.Labels[targetedNodeLabel] = ""
```

Comment:
do we need to propagate this label to machineSet.Spec.Template.Spec.Label as well?

Reply:
that's actually what this is doing, https://github.com/openshift/cluster-api-actuator-pkg/pull/140/files#diff-c7d1003b84db4b5057508a81ab203c46L66

Reply:
that's what i thought when i looked through the code. i was confused because i had _thought_ those labels would be copied by the autoscaler.